src/UNL/Peoplefinder/Driver/OracleDB.php

```diff
@@ -31,6 +31,18 @@ private function closeConnection()
 
 	public function query($statement, $params = array())
 	{
+	    $cache = $this->getCache();
+	    
+	    //Use md5 so we don't exceed the memcached key length
+	    $cache_key = 'oracle_query_' .  md5($statement) . '--' . md5(serialize($params));
+
+        if ($result = $cache->get($cache_key)) {
+            $result = unserialize($result);
+            return $result;
+        }
+
+        $cache->set($cache_key, serialize([]));
```

Comment:
The idea being that only the first request will hit the database. If 100 people request the same resource at the same time, there would be 100 requests to the database. With this method, other 99 would just see an empty result until they refresh the page. It was necessary with LDAP because some LDAP requests could take several seconds, and then overload the LDAP server. I'm not so sure it is as necessary here. My opinion is to err on the side of caution.